src/ucp/rma/rma_send.c

```diff
@@ -259,17 +252,20 @@ ucs_status_ptr_t ucp_put_nbx(ucp_ep_h ep, const void *buffer, size_t count,
             goto out_unlock;
         }
 
+        datatype = ucp_request_param_datatype(param);
         req = ucp_request_get_param(worker, param,
                                     {ret = UCS_STATUS_PTR(UCS_ERR_NO_MEMORY);
                                     goto out_unlock;});
         req->send.rma.rkey        = rkey;
         req->send.rma.remote_addr = remote_addr;
+        if (UCP_DT_IS_CONTIG(datatype)) {
```

Comment:
can reduce one branch here: try to do something similar to https://github.com/openucx/ucx/blob/3c35b028a0fb1553f321d0ff727cc78757e2f30e/src/ucp/tag/tag_send.c#L251
even unite common code with it

Reply:
Not sure I understood correctly. I did add a branch in the code, but it is supposed to improve in runtime (due to `ucs_likely`).